hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/FileIOEngine.java

```diff
@@ -267,6 +339,75 @@ void refreshFileConnection(int accessFileNum, IOException ioe) throws IOExceptio
     }
   }
 
+  /**
+   * Delete bucketcache files
+   */
+  private void deleteCacheDataFile() {
+    if (filePaths == null) {
+      return;
+    }
+    for (String file : filePaths) {
+      new File(file).delete();
+    }
+  }
+
+  @Override
+  public byte[] calculateChecksum()
+    throws IOException, NoSuchAlgorithmException {
+    if (filePaths == null) {
+      return null;
+    }
+    StringBuilder sb = new StringBuilder();
+    for (String filePath : filePaths){
+      File file = new File(filePath);
+      if (file.exists()){
+        sb.append(filePath);
+        sb.append(getFileSize(filePath));
+        sb.append(file.lastModified());
+      } else {
+        throw new IOException("Cache file: " + filePath + " is not exists.");
+      }
+    }
+    MessageDigest messageDigest = MessageDigest.getInstance(algorithmName);
+    messageDigest.update(Bytes.toBytes(sb.toString()));
+    return messageDigest.digest();
+  }
+
+  @Override
+  public boolean isOldVersion() {
```

Comment:
I see now this is been used.   So my point was like read the checksum in BucketCache only. And let IT decide whether to do the checksum verify or not based on PBMagic.

Reply:
I got your point.